pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2015,15 +2116,71 @@ func xlUnpause(domainName string, domainId int) error {
 	if err != nil {
 		log.Errorln("xl unpause failed ", err)
 		log.Errorln("xl unpause output ", string(stdoutStderr))
-		return errors.New(fmt.Sprintf("xl unpause failed: %s\n",
-			string(stdoutStderr)))
+		return fmt.Errorf("xl unpause failed: %s\n",
+			string(stdoutStderr))
 	}
 	log.Infof("xlUnpause done. Result %s\n", string(stdoutStderr))
 	return nil
 }
 
-func xlShutdown(domainName string, domainId int, force bool) error {
-	log.Infof("xlShutdown %s %d\n", domainName, domainId)
+// DomainShutdown is a wrapper for domain shutdown thru xlShutdown or rktStop
+func DomainShutdown(status types.DomainStatus, force bool) error {
+
+	var err error
+	log.Infof("DomainShutdown force-%v %s %d\n", force, status.DomainName, status.DomainId)
+
+	if status.IsContainer {
+		// Use rkt tool
+		log.Infof("Using rkt tool ... PodUUID - %s\n", status.PodUUID)
+		err = rktStop(status.PodUUID, force)
+		if err != nil {
+			log.Errorf("rktStop %s failed: %s\n", status.DomainName, err)
+		}
+		// Unconditionally invoke xl shutdown
+		log.Infof("Invoke xl tool ... DomainName - %s\n", status.DomainName)
+		err = xlShutdown(status.DomainName, status.DomainId, force)
+	} else {
+		// Use xl tool
+		log.Infof("Using xl tool ... DomainName - %s\n", status.DomainName)
+		err = xlShutdown(status.DomainName, status.DomainId, force)
+	}
+
+	return err
+}
+
+func rktStop(PodUUID string, force bool) error {
```

Comment:
This and the rm below seems to effect all of the containers running on the pod, and there seems to be one pod per device. That would be wrong. Should only stop/rm the particular app instance.

How can that be done?

Reply:
Actually this is not the case @nordmark -- in rkt the pods == containers. These are NOT Kubernetes like pods.

Hence what @spatchineelam is doing here is fine as per rkt design